apps/communications/contacts/test/unit/views/matching/matching_controller_test.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+/* global matchingDetailsData */
+/* global MockMozL10n */
+/* global MockCurtain */
+/* global MockContactsService */
+/* global LazyLoader */
+/* global Curtain */
+/* global MatchingController */
+/* global Matcher */
+/* global ContactsService */
+/* global MergeHelper */
+
+require('/shared/js/lazy_loader.js');
+
+requireApp('communications/contacts/test/unit/mock_l10n.js');
+requireApp('communications/contacts/test/unit/mock_matching_contacts.html.js');
+requireApp('communications/contacts/test/unit/' +
+           'contacts_matching_ui_test_data.js');
+requireApp('communications/contacts/test/unit/mock_contacts_service.js');
+requireApp('communications/facebook/test/unit/mock_curtain.js');
+
+requireApp('communications/contacts/views/matching/js/matching_controller.js');
+
+
+suite('MatchingController', function() {
+
+  var realL10n, realCurtain, realContactsService, realMatcher, realMergeHelper;
+
+  suiteSetup(function() {
+    realL10n = window.navigator.mozL10n;
+    window.navigator.mozL10n = MockMozL10n;
+    realCurtain = window.Curtain;
+    window.Curtain = MockCurtain;
+    realContactsService = window.ContactsService;
+    window.ContactsService = MockContactsService;
+    realMatcher = window.Matcher;
+    window.Matcher = {
+      match: function foo(){}
+    };
+    realMergeHelper = window.MergeHelper;
+    window.MergeHelper = {
+      merge: function foo() {
+        return {
+          then: function boo(cb) {
+            cb();
+          }
+        };
+      }
+    };
+  });
+
+  suiteTeardown(function() {
+    window.navigator.mozL10n = realL10n;
+    realL10n = null;
+
+    window.Curtain = realCurtain;
+    realCurtain = null;
+
+    window.ContactsService = realContactsService;
+    realContactsService = null;
+
+    window.Matcher = realMatcher;
+    realMatcher = null;
+
+    window.MergeHelper = realMergeHelper;
+    realMergeHelper = null;
+  });
+
+  setup(function() {
+    this.sinon.stub(LazyLoader, 'load', function(files, cb) {
+      cb();
+    });
+  });
+
+  suite(' > On merge', function() {
+    var showSpy, hideMenuSpy, mergeSpy, hideSpy;
+    setup(function() {
+      showSpy = this.sinon.spy(Curtain, 'show');
+      hideMenuSpy = this.sinon.spy(Curtain, 'hideMenu');
+      hideSpy = this.sinon.spy(Curtain, 'hide');
+      mergeSpy = this.sinon.spy(MergeHelper, 'merge');
+
+      // Hack
+      parent.ContactsService = window.ContactsService;
+
+      MatchingController.init();
+    });
+
+    teardown(function() {
+    });
+
+    test('On merge event received', function() {
+      window.dispatchEvent(new CustomEvent('merge', {
+        'detail': {
+          'checkedContacts': {'1a': '1a'}
+        }
+      }));
+      assert.isTrue(showSpy.called);
+      assert.isTrue(hideMenuSpy.called);
+      assert.isTrue(mergeSpy.called);
+      assert.isTrue(hideSpy.called);
```

Comment:
Do you know how many times all these functions must be called? Probably it would be a more accurate test specifying it.